### PR TITLE
Sky completed the jasmine basics assessment

### PR DESCRIPTION
**Pull request by skywinston**
My solution for `modeScore` is kinda crazy.  I'm really curious if there's another way to do this easily with higher order functions.  I look forward to discussing other solution next week.

### CODE REVIEW

File: grader.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    return result;
+  },
+
+  modeScore: function(array){
+    var modeMap = {};
+    var max = 0;
+    var mode = 0;
+    var output;
+    array.forEach(function(score){
+        if(modeMap[score] == null){     // if score is not found in modeMap object...
```
Comment (craftninja) [inline grader.js]: Is `null` truthy or falsy?

File: grader.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            modeMap[score] = 1;         // add the score as a property and initialize its value at 1
+        } else {                        // otherwise, this score already exists, in which case...
+            modeMap[score]++;           // increment its "count" value.
+        }
```
Comment (craftninja) [inline grader.js]: Could this be effectively onelined?

File: grader.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    for (var property in modeMap){     // loop through properties again...
+        if(modeMap[property] === mode){// if the value of the current property iteration equals the mode...
+            output = property;         // set the output equal to that property (score).   
+        }
+    }
+    return Number(output);             // return the mode score and force it to a number.
+  }
+
+}
```
Comment (craftninja) [inline grader.js]: Comments are often considered a "code smell", or refactoring flag. If you need comments to understand what is going on, consider pulling out well named variables and functions to help clarify. Tests all pass, Great Job!

File: spec/spec.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  it('takes an even array of test scores and returns the average of the middle values', function(){
+    expect(code.medianScore([23,45,67,89,96,99])).toEqual(78);
+    expect(code.medianScore([1,4,7,9,12,15,25,46])).toEqual(10.5);
+  });
+});
+
+describe('modeScore', function(){
+  it('takes an array of numbers and returns the number that appears most', function(){
+    expect(code.modeScore([82, 92, 75, 91, 92, 89, 95, 100, 86])).toEqual(92);
+    expect(code.modeScore([82, 92, 75, 91, 92, 89, 95, 100, 86, 86, 86])).toEqual(86);
```
Comment (craftninja) [inline spec/spec.js]: For long arrays, strings or objects in your tests, feel free to pull that out into a variable.

**Comment by craftninja** [comment]
All tests pass. Nice Job!

**Comment by skywinston** [comment]
Hey Em,

Thanks for the comments.  Will the instructor team be expecting us to refactor these problems based on the feedback you provide, or is the feedback simply to provoke thought and build understanding?

**Pull request closed without merging**